src/pages/index.js

```diff
@@ -124,23 +161,109 @@ function handleEscapeKey(evt) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editProfileModalNameInput.value;
-  profileDescription.textContent = editProfileModalDescriptionInput.value;
-  closeModal(editProfileModal);
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editProfileModalNameInput.value,
+      about: editProfileModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+
   evt.target.reset();
 }
 
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = {
-    name: cardModalNameInput.value,
-    link: cardModalLinkInput.value,
-  };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  closeModal(editCardModal);
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  api
+    .addNewCard({
+      name: cardModalNameInput.value,
+      link: cardModalLinkInput.value,
+    })
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      closeModal(editCardModal);
+      evt.target.reset();
+      disableButton(cardSubmitBtn);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleDeleteCardSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Deleting...";
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deteleCardModal);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitBtn.textContent = "Delete";
+    });
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deteleCardModal);
+}
+
+function handleAvatarFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+
+  api
+    .editAvatar({
+      avatar: avatarInput.value,
+    })
+    .then((data) => {
+      document.querySelector(".profile__picture").src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+
   evt.target.reset();
-  disableButton(cardSubmitBtn, settings);
+  disableButton(avatarSubmitBtn);
+}
+
+function disableButton(button) {
+  if (!settings || !settings.inactiveButtonClass) {
+    console.error("Settings object or inactiveButtonClass is missing");
+    return;
+  }
+
+  button.classList.add(settings.inactiveButtonClass);
+  button.disabled = true;
```

Comment:
Please, delete `disableButton` from `index.js` because you have it inside` validation.js`. You should **export** `disableButton` from there, to be able to **import** it into `index.js`